core/runtime/src/main/java/io/quarkus/runtime/StartupContext.java

```diff
@@ -20,26 +20,26 @@ public class StartupContext implements Closeable {
     // For example, the raw command line args and ShutdownContext are set when the StartupContext is created
     private final Map<String, Object> values = new HashMap<>();
 
-    private final Deque<Runnable> shutdownTasks = new ConcurrentLinkedDeque<>();
-    private final Deque<Runnable> lastShutdownTasks = new ConcurrentLinkedDeque<>();
+    private final PriorityBlockingQueue<ShutdownTask> shutdownTasks = new PriorityBlockingQueue<>();
+    private final PriorityBlockingQueue<ShutdownTask> lastShutdownTasks = new PriorityBlockingQueue<>();
```

Comment:
From the javadoc of `PriorityBlockingQueue`:

> Operations on this class make no guarantees about the ordering of elements with equal priority

This looks problematic for backwards compatibility, where all tasks will share the same "default" priority, but will (probably) be added in a specific order that just happens to work?

FWIW they also provide a solution in that same javadoc:

> If you need to enforce an ordering, you can define custom classes or comparators that use a secondary key to break ties in primary priority values.